src/__tests__/persistCache.ts

```diff
@@ -0,0 +1,155 @@
+import gql from 'graphql-tag';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { Hermes } from 'apollo-cache-hermes';
+
+import { persistCache } from '../';
+import MockStorage from '../__mocks__/MockStorage';
+import { simulateApp, simulateWrite } from '../__mocks__/simulate';
+
+jest.useFakeTimers();
+describe('persistCache', () => {
+  describe('setup', () => {
+    it('requires a cache', async () => {
+      try {
+        await persistCache({ storage: new MockStorage() });
+        fail('invoking persistCache without a cache should throw an error');
+      } catch (e) {
+        expect(() => {
+          throw e;
+        }).toThrowErrorMatchingSnapshot();
+      }
```

Comment:
Jest 2.0 added a simpler pattern for dealing with promise rejections. I think you can just do this now:

```js
it ('requires a cache', async () => {
  await expect(persistCache({ storage: new MockStorage() })).rejects.toThrowErrorMatchingSnapshot();
});
```

More here: https://facebook.github.io/jest/docs/en/expect.html#rejects

Reply:
ooooh this is much cleaner 😍 thanks for filling me in!

Reply:
just tried this and it doesn't work 😢 I think maybe it has to [specifically catch the error message](https://facebook.github.io/jest/docs/en/tutorial-async.html#error-handling) to work?